src/main/kotlin/org/rust/lang/core/types/infer/Expressions.kt

```diff
@@ -127,10 +127,24 @@ fun inferExpressionType(expr: RsExpr): RustType {
                 RustUnknownType
         }
 
+        is RsArrayExpr -> inferArrayType(expr)
+
         else -> RustUnknownType
     }
 }
 
+private fun inferArrayType(expr: RsArrayExpr): RustType {
+    val (elementType, size) = if (expr.semicolon != null) {
+        if (expr.exprList.size != 2) return RustUnknownType
```

Comment:
Generally, I'm not sure that this check is necessary here because this is contrary to grammar https://github.com/intellij-rust/intellij-rust/blob/c495d7caf2e7272e80cf53fc3dcf5c1e49f144dc/src/main/grammars/RustParser.bnf#L922.
Maybe drop it at all?